### PR TITLE
Handle errors triggered by login event handlers

### PR DESCRIPTION
**Pull request by cocoke**
Prevent user login if any error is triggered by a login event handler (like ReCaptcha)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=8899) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=8899) before we can accept your contribution.<br/><br/>:x: cocoke [sign now](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=8899)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OrchardCMS/OrchardCore?pullRequest=8899) it.</sub>

File: src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs [hunk 1]
```diff
@@ -194,29 +194,32 @@ public async Task<IActionResult> Login(LoginViewModel model, string returnUrl =
                 else
                 {
                     await _accountEvents.InvokeAsync((e, model, modelState) => e.LoggingInAsync(model.UserName, (key, message) => modelState.AddModelError(key, message)), model, ModelState, _logger);
-                    var user = await _userManager.FindByNameAsync(model.UserName) ?? await _userManager.FindByEmailAsync(model.UserName);
-                    // This doesn't count login failures towards account lockout
-                    // To enable password failures to trigger account lockout, set lockoutOnFailure: true
-                    if (user != null)
+                    if(ModelState.IsValid)
```
Comment (agriffard) [inline src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs:197]: Please format document.

Comment (cocoke) [reply src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs]: Done ✔

File: src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs [hunk 2]
```diff
@@ -194,29 +194,32 @@ public async Task<IActionResult> Login(LoginViewModel model, string returnUrl =
                 else
                 {
                     await _accountEvents.InvokeAsync((e, model, modelState) => e.LoggingInAsync(model.UserName, (key, message) => modelState.AddModelError(key, message)), model, ModelState, _logger);
-                    var user = await _userManager.FindByNameAsync(model.UserName) ?? await _userManager.FindByEmailAsync(model.UserName);
-                    // This doesn't count login failures towards account lockout
-                    // To enable password failures to trigger account lockout, set lockoutOnFailure: true
-                    if (user != null)
+                    if (ModelState.IsValid)
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs:197]: Ok doesn't hurt.

**Review by Skrypt — APPROVED** [review]
Only adds an extra validation on ModelState.IsValid which makes sense to me.

**Comment by agriffard** [comment]
@all-contributors please add cocoke for code

**Comment by cocoke** [comment]
> @cocoke Did you have issues with the current Recaptcha in your login?

Yes, I've made a login/register handler that require a custom token to log in but users were able to log in even after an error is triggered. So I tested the ReCaptcha module and I was also able to log in without verification.

**Pull request merged**